entrypoint.sh

```diff
@@ -7,227 +7,247 @@ set -o pipefail
+	fi
+
+	linter_failed="false"
+	declare -a linter_failures
+	declare -a linter_successes
 }
 
 function check_environment() {
-  # Check the GITHUB_BASE_REF (PRs only)
-  if [[ "${GITHUB_ACTIONS:-false}" == "true" && -n ${GITHUB_BASE_REF:+x} ]]; then
-    mainline="${GITHUB_BASE_REF##*/}"
-    if [[ "${mainline}" != "main" ]]; then
-      feedback ERROR "Base branch name is not main"
-    fi
-  fi
-
-  # Ensure dictionaries don't have overlap
-  overlap=$(comm -12 <(sort "${GLOBAL_DICTIONARY}" | tr '[:upper:]' '[:lower:]') \
-                     <(sort "${REPO_DICTIONARY}"   | tr '[:upper:]' '[:lower:]'))
-  if [[ "${overlap}" ]]; then
-    feedback WARNING "The following words are already in the global dictionary:
+	# Check the GITHUB_BASE_REF (PRs only)
+	if [[ ${GITHUB_ACTIONS:-false} == "true" && -n ${GITHUB_BASE_REF:+x} ]]; then
+		mainline="${GITHUB_BASE_REF##*/}"
+		if [[ ${mainline} != "main" ]]; then
+			feedback ERROR "Base branch name is not main"
+		fi
+	fi
+
+	# Ensure dictionaries don't have overlap
+	overlap=$(comm -12 <(sort "${GLOBAL_DICTIONARY}" | tr '[:upper:]' '[:lower:]') \
+		<(sort "${REPO_DICTIONARY}" | tr '[:upper:]' '[:lower:]'))
+	if [[ "${overlap}" ]]; then
+		feedback WARNING "The following words are already in the global dictionary:
 ${overlap}"
-    feedback ERROR "Overlap was detected in the per-repo and global dictionaries"
-  fi
+		feedback ERROR "Overlap was detected in the per-repo and global dictionaries"
+	fi
 }
 
-function super_lint() {
-  superlinter_logfile="/opt/goat/log/super-linter.log"
+function detect_kubernetes_file() {
+	local file="$1"
 
-  echo -e "\nRunning Super-Linter\n--------------------------\n"
-  echo "===============================" >> "$superlinter_logfile"
-  echo "SUPER-LINTER" >> "$superlinter_logfile"
+	if grep -q -v 'kustomize.config.k8s.io' "${file}" &&
+		grep -q -v "tekton" "${file}" &&
+		grep -q -E '(apiVersion):' "${file}" &&
+		grep -q -E '(kind):' "${file}"; then
+		return 0
+	fi
 
-  # Turn off the possum
-  export SUPPRESS_POSSUM="true"
-
-  if [[ -n ${GITHUB_WORKSPACE:-} ]]; then
-    echo "Setting ${GITHUB_WORKSPACE} as safe directory"
-    git config --global --add safe.directory "${GITHUB_WORKSPACE}"
+	return 1
+}
 
-  fi
+function detect_cloudformation_file() {
+	local file="$1"
 
-  # When run in a pipeline, move per-repo configurations into the right location at runtime so super-linter finds them, overwriting the defaults.
-  # This will handle hidden and non-hidden files, as well as sym links
-  cp -p "${GITHUB_WORKSPACE:-.}/.github/linters/"* "${GITHUB_WORKSPACE:-.}/.github/linters/".* /etc/opt/goat/ || true
-  
-  set +e
-  superlinter_result=$(/action/lib/linter.sh >> "$superlinter_logfile" 2>&1; echo $?)
-  set -e
+	if grep -q 'AWSTemplateFormatVersion' "${file}" >/dev/null; then
+		return 0
+	fi
 
-  echo "-------------------------------" >> "$superlinter_logfile"
+	if grep -q -E '(AWS|Alexa|Custom)::' "${file}" >/dev/null; then
```

Comment:
Please add a comment to explain why this is here

Reply:
Done